src/messaging/Message.utils.ts

```diff
@@ -0,0 +1,300 @@
+          : undefined,
+        body.content.prerequisiteClaims
+          ? body.content.prerequisiteClaims.map((claim) =>
+              ClaimUtils.compress(claim)
+            )
+          : undefined,
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.SUBMIT_ATTESTATION_FOR_CLAIM: {
+      const compressedContents: CompressedSubmitAttestationForClaimContent = [
+        AttestationUtils.compress(body.content.attestation),
+        body.content.attestationPE || undefined,
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.REQUEST_CLAIMS_FOR_CTYPES: {
+      const compressedContents: CompressedRequestClaimsForCTypesContent = [
+        body.content.ctypes,
+        body.content.peRequest,
+        body.content.allowPE,
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.SUBMIT_CLAIMS_FOR_CTYPES_CLASSIC: {
+      const compressedContents: CompressedAttestedClaim[] = body.content.map(
+        (attestedClaim: IAttestedClaim | CompressedAttestedClaim) =>
+          Array.isArray(attestedClaim)
+            ? attestedClaim
+            : AttestedClaimUtils.compress(attestedClaim)
+      )
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.REQUEST_ACCEPT_DELEGATION: {
+      const compressedContents: CompressedRequestDelegationApproval = [
+        [
+          body.content.delegationData.account,
+          body.content.delegationData.id,
+          body.content.delegationData.parentId,
+          body.content.delegationData.permissions,
+          body.content.delegationData.isPCR,
+        ],
+        body.content.signatures.inviter,
+        body.content.metaData,
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.SUBMIT_ACCEPT_DELEGATION: {
+      const compressedContents: CompressedSubmitDelegationApproval = [
+        [
+          body.content.delegationData.account,
+          body.content.delegationData.id,
+          body.content.delegationData.parentId,
+          body.content.delegationData.permissions,
+          body.content.delegationData.isPCR,
+        ],
+        [body.content.signatures.inviter, body.content.signatures.invitee],
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.REJECT_ACCEPT_DELEGATION: {
+      const compressedContents: CompressedDelegationData = [
+        body.content.account,
+        body.content.id,
+        body.content.parentId,
+        body.content.permissions,
+        body.content.isPCR,
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.INFORM_CREATE_DELEGATION: {
+      const compressedContents: CompressedInformDelegationCreation = [
+        body.content.delegationId,
+        body.content.isPCR,
+      ]
+      return [body.type, compressedContents]
+    }
+    default:
+      throw SDKErrors.ERROR_MESSAGE_BODY_MALFORMED()
+  }
+}
```

Comment:
because I tried it out, in case you want to use it you can just commit from the comment:
```suggestion
export function compressMessage(body: MessageBody): CompressedMessageBody {
  let compressedContents: CompressedMessageBody[1]
  switch (body.type) {
    case MessageBodyType.REQUEST_TERMS: {
      compressedContents = ClaimUtils.compress(body.content)
      break;
    }
    case MessageBodyType.SUBMIT_TERMS: {
      compressedContents = [
        ClaimUtils.compress(body.content.claim),
        body.content.legitimations.map(
          (attestedClaim: IAttestedClaim | CompressedAttestedClaim) =>
            Array.isArray(attestedClaim)
              ? attestedClaim
              : AttestedClaimUtils.compress(attestedClaim)
        ),
        body.content.delegationId,
        body.content.quote
          ? QuoteUtils.compressAttesterSignedQuote(body.content.quote)
          : undefined,
        body.content.prerequisiteClaims,
      ]
      break;
    }
    case MessageBodyType.REJECT_TERMS: {
      compressedContents = [
        ClaimUtils.compress(body.content.claim),
        body.content.legitimations.map((val) =>
          AttestedClaimUtils.compress(val)
        ),
        body.content.delegationId || undefined,
      ]
      break;
    }
    case MessageBodyType.REQUEST_ATTESTATION_FOR_CLAIM: {
      compressedContents = [
        RequestForAttestationUtils.compress(body.content.requestForAttestation),
        body.content.quote
          ? QuoteUtils.compressQuoteAgreement(body.content.quote)
          : undefined,
        body.content.prerequisiteClaims
          ? body.content.prerequisiteClaims.map((claim) =>
              ClaimUtils.compress(claim)
            )
          : undefined,
      ]
      break;
    }
    case MessageBodyType.SUBMIT_ATTESTATION_FOR_CLAIM: {
      compressedContents = [
        AttestationUtils.compress(body.content.attestation),
        body.content.attestationPE || undefined,
      ]
      break;
    }
    case MessageBodyType.REQUEST_CLAIMS_FOR_CTYPES: {
      compressedContents = [
        body.content.ctypes,
        body.content.peRequest,
        body.content.allowPE,
      ]
      break;
    }
    case MessageBodyType.SUBMIT_CLAIMS_FOR_CTYPES_CLASSIC: {
      compressedContents = body.content.map(
        (attestedClaim: IAttestedClaim | CompressedAttestedClaim) =>
          Array.isArray(attestedClaim)
            ? attestedClaim
            : AttestedClaimUtils.compress(attestedClaim)
      )
      break;
    }
    case MessageBodyType.REQUEST_ACCEPT_DELEGATION: {
      compressedContents = [
        [
          body.content.delegationData.account,
          body.content.delegationData.id,
          body.content.delegationData.parentId,
          body.content.delegationData.permissions,
          body.content.delegationData.isPCR,
        ],
        body.content.signatures.inviter,
        body.content.metaData,
      ]
      break;
    }
    case MessageBodyType.SUBMIT_ACCEPT_DELEGATION: {
      compressedContents = [
        [
          body.content.delegationData.account,
          body.content.delegationData.id,
          body.content.delegationData.parentId,
          body.content.delegationData.permissions,
          body.content.delegationData.isPCR,
        ],
        [body.content.signatures.inviter, body.content.signatures.invitee],
      ]
      break;
    }
    case MessageBodyType.REJECT_ACCEPT_DELEGATION: {
      compressedContents = [
        body.content.account,
        body.content.id,
        body.content.parentId,
        body.content.permissions,
        body.content.isPCR,
      ]
      break;
    }
    case MessageBodyType.INFORM_CREATE_DELEGATION: {
      compressedContents = [
        body.content.delegationId,
        body.content.isPCR,
      ]
      break;
    }
    default:
      throw SDKErrors.ERROR_MESSAGE_BODY_MALFORMED()
  }
   return [body.type, compressedContents] as CompressedMessageBody
}
```

Reply:
I tried this way before, but the issue was using `as`. I don't know if we should use it. I was told not too.

Though it seems we are using it in the code.